optimum/habana/diffusers/models/unet_2d.py

```diff
@@ -0,0 +1,112 @@
+from typing import Optional, Tuple, Union
+
+import torch
+from diffusers.models.unets.unet_2d import UNet2DOutput
+
+from optimum.utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+
+def gaudi_unet_2d_model_forward(
+    self,
+    sample: torch.FloatTensor,
+    timestep: Union[torch.Tensor, float, int],
+    cal_timestep_only: bool = True,
+    emb: Optional[torch.Tensor] = None,
+    class_labels: Optional[torch.Tensor] = None,
+    return_dict: bool = True,
+) -> Union[UNet2DOutput, Tuple]:
+    r"""
+
+    Copied from: https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/unets/unet_2d.py#L243
+
+    Changes:
+        1. Add an if case to separate timestep calculations and the diffusion process
+    """
+    # 0. center input if necessary
+    if self.config.center_input_sample:
+        sample = 2 * sample - 1.0
+
+    # Run on CPU
```

Comment:
Please update this this is not running on CPU, but running on HPU with fp32.

Reply:
Also instead of dividing code like this, and passing two additional argument,  you can simply disable the autocast just for the timesteps section in the model code as below. 
        with torch.autocast(device_type="hpu", enabled=False):

Reply:
> Also instead of dividing code like this, and passing two additional argument, you can simply disable the autocast just for the timesteps section in the model code as below. with torch.autocast(device_type="hpu", enabled=False):

Hey jimin,
I have attempted this before and it did not work well in graph mode. Eventually, I have to separate it into 2 parts.

Reply:
> Please update this, this is not running on CPU, but running on HPU with fp32.

Updated the comment. It now say that the code is running on Gaudi using FP32.